runtime/Java/src/org/antlr/v4/runtime/atn/ATNState.java

```diff
@@ -187,7 +187,7 @@ public void addTransition(int index, Transition e) {
 			epsilonOnlyTransitions = e.isEpsilon();
 		}
 		else if (epsilonOnlyTransitions != e.isEpsilon()) {
-			System.err.format(Locale.getDefault(), "ATN state %d has both epsilon and non-epsilon transitions.\n", stateNumber);
+			System.err.format(Locale.getDefault(), "ATN state %d has both epsilon and non-epsilon transitions.%n", stateNumber);
```

Comment:
This should be impossible to hit. It's mostly here to make sure that if a code change is made (e.g. a tree-based optimization algorithm for grammars) which breaks this condition, it gets reported.